src/sentry/utils/samples.py

```diff
@@ -301,10 +301,14 @@ def create_sample_event(
             data["contexts"]["trace"][key] = kwargs.pop(key)
 
     data.update(kwargs)
-    return create_sample_event_basic(data, project.id, raw=raw)
+    return create_sample_event_basic(data, project.id, raw=raw, tagged=True)
 
 
-def create_sample_event_basic(data, project_id, raw=True, skip_send_first_transaction=False):
+def create_sample_event_basic(
```

Comment:
@Neo-Zhixing instead of adding a new argument, can't we just modify `data` as we pass it in?

Reply:
Sure, but then you'll have to do this from two places

Reply:
@Neo-Zhixing that's fine, we can just make a function to do it so we have less repeating code

Reply:
Both from `create_sample_event`, and the create transaction endpoint, because the create transaction endpoint calls `create_sample_event_basic` while the create error endpoint calls `create_sample_event`.

Reply:
@Neo-Zhixing For transactions, we have a function called `fix_event_data` which is the best place to make modifications for sample event info. We don't have something for sample errors but maybe with refactoring, we can make it work in a similar way.

Reply:
To me the `sample_event` tag is clearly coupled with the fact that it was created by `create_sample_event_basic`. So it makes sense to do this in `create_sample_event_basic` as an argument. `fix_event_data` does one thing which is fixing event data, but with the `sample_event` flag you're not fixing things, you're adding onto the data.